beacon-chain/forkchoice/doubly-linked-tree/unrealized_justification.go

```diff
@@ -56,17 +58,26 @@ func (f *ForkChoice) updateUnrealizedCheckpoints() {
 		if node.justifiedEpoch > f.store.justifiedCheckpoint.Epoch {
 			f.store.prevJustifiedCheckpoint = f.store.justifiedCheckpoint
 			f.store.justifiedCheckpoint = f.store.unrealizedJustifiedCheckpoint
+			if err := f.updateJustifiedBalances(ctx, f.store.justifiedCheckpoint.Root); err != nil {
+				return errors.Wrap(err, "could not update justified balances")
+			}
 			if !features.Get().EnableDefensivePull && node.justifiedEpoch > f.store.bestJustifiedCheckpoint.Epoch {
 				f.store.bestJustifiedCheckpoint = f.store.unrealizedJustifiedCheckpoint
 			}
 		}
 		if node.finalizedEpoch > f.store.finalizedCheckpoint.Epoch {
 			if !features.Get().EnableDefensivePull {
-				f.store.justifiedCheckpoint = f.store.unrealizedJustifiedCheckpoint
+				if f.store.unrealizedJustifiedCheckpoint.Epoch != f.store.justifiedCheckpoint.Epoch {
+					if err := f.updateJustifiedBalances(ctx, f.store.unrealizedJustifiedCheckpoint.Root); err != nil {
+						return errors.Wrap(err, "could not update justified balances")
+					}
+					f.store.justifiedCheckpoint = f.store.unrealizedJustifiedCheckpoint
```

Comment:
Previously the update `f.store.justifiedCheckpoint = f.store.unrealizedJustifiedCheckpoint` did not rely on `if f.store.unrealizedJustifiedCheckpoint.Epoch != f.store.justifiedCheckpoint.Epoch`. Shouldn't the update be done outside the `if` statement?